projects/centre_console/src/drive_fsm.c

```diff
@@ -123,8 +125,12 @@ static void prv_set_motorcontroller_output(Fsm *fsm, const Event *e, void *conte
   DriveFsmStorage *storage = (DriveFsmStorage *)context;
   storage->current_state = DRIVE_STATE_TRANSITIONING;
   DestinationTransitionInfo info = s_destination_transition_lookup[storage->destination];
-  DriveFsmFault fault = { .fault_reason = DRIVE_FSM_FAULT_REASON_MCI_OUTPUT,
-                          .fault_state = info.mci_drive_output };
+  StateTransitionFault fault = {
+    .state_machine = DRIVE_FSM_STATE_MACHINE,
+    .fault_reason = (DriveFsmFaultReason){ .step = DRIVE_FSM_TRANSITION_STEP_MCI_OUTPUT,
+                                           .state = info.mci_drive_output }
+                        .raw
```

Comment:
Is the `(DriveFsmFaultReason)` necessary here? I don't think I've seen that anywhere else.
Same with the `.raw`

Reply:
Similar style thing used below, so I guess if it isn't then change those too